common/src/playground_util/pause.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _PLAYGROUND_PAUSE_H
+#define _PLAYGROUND_PAUSE_H
+
+#include <stddef.h>
+
+// Busy wait a short while. Typically used to avoid overflowing UART send
+// buffers
+inline static void pause() {
```

Comment:
Have you considered using `busy_wait(unsigned)` or `busy_wait_us(unsigned)` ?   
https://github.com/enjoy-digital/litex/blob/master/litex/soc/software/libbase/system.c#L18-L36
https://github.com/enjoy-digital/litex/blob/master/litex/soc/cores/cpu/vexriscv/system.h#L39-L40

I suppose the question is whether every board will have timer0.   The solution you have in the PR is self-contained and portable.

Reply:
IIRC, the issue is that the header files does not work in C++.